### PR TITLE
Update `QNGOptimizer` to handle deprecations

### PR DESCRIPTION
**Pull request by antalszava**
**Context**

The `QNGOptimizer` raises deprecation warnings because (see [related demo](http://pennylane.ai-dev.s3-website-us-east-1.amazonaws.com/qml/demos/tutorial_quantum_natural_gradient.html#quantum-natural-gradient-optimization)):

* inputs have to explicitly specify requires_grad=True;
* the keyword argument diag_approx is deprecated.

**Changes**

* Switches to using PennyLane NumPy for the optimizer
* Switches how the optimizer uses `qml.metric_tensor`
* Adds a deprecation warning when `diag_approx` is specified with `QNGOptimizer`

### CODE REVIEW

File: pennylane/optimize/qng.py
Comment (josh146) [inline pennylane/optimize/qng.py:199]: @antalszava do you think we should update the QNG optimizer to match the change in the metric tensor?

That is,

```python
class QNGOptimizer:

    def __init__(self, stepsize=0.01, approx="diag", lam=0):
```

?

I had forgotten the QNG optimizer also used this old terminology! I think it makes sense to match the metric tensor here, that way we can simply have

```suggestion
                metric_tensor_fn = qml.metric_tensor(qnode, approx=self.approx)
```

**Review by josh146 — COMMENTED** [review]
Thanks Antal for following up on this one! 💪 💯 

I have a quick question though before approval.

I tried running the following script on master, and received no warning:

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=1)

@qml.qnode(dev)
def circuit(params):
    qml.RX(params[0], wires=0)
    qml.RY(params[1], wires=0)
    return qml.expval(qml.PauliX(0) + qml.PauliZ(0))

theta = np.array([0.011, 0.012], requires_grad=True)

opt = qml.QNGOptimizer(0.01)
new_params = opt.step(circuit, theta)

res = opt.metric_tensor
exp = np.diag([0.25, (np.cos(theta[0]) ** 2) / 4])
assert np.allclose(res, exp)
```

This is based on the test you modified here.

After having a look at the demo you linked, I think the issue is with the _demo_, not the `QNGOptimizer`! If you have a look at the demo, it is importing vanilla NumPy, and not setting `requires_grad=True` 🙂

File: pennylane/optimize/qng.py
Comment (josh146) [inline pennylane/optimize/qng.py:205]:
```suggestion
        new_args = np.array(self.apply_grad(g, args), requires_grad=True)
```

better I think to always be explicit moving forward!

**Review by josh146 — APPROVED** [review]
@antalszava nevermind, I see that an updated version of the demo _does_ use the pennylane numpy! https://4409-214003948-gh.circle-artifacts.com/0/_build/html/demos/tutorial_quantum_natural_gradient.html I am stumped as to why I cannot reproduce it locally 🤔

**Comment by antalszava** [comment]
Hi @josh146 thank you :slightly_smiling_face: The warnings related to the `requires_grad=True` casting seem to arise if we step the optimizer once more:

```python
import pennylane as qml
from pennylane import numpy as np

dev = qml.device("default.qubit", wires=1)

@qml.qnode(dev)
def circuit(params):
    qml.RX(params[0], wires=0)
    qml.RY(params[1], wires=0)
    return qml.expval(qml.PauliX(0) + qml.PauliZ(0))

theta = np.array([0.011, 0.012], requires_grad=True)

opt = qml.QNGOptimizer(0.01)
new_params = opt.step(circuit, theta)

res = opt.metric_tensor
exp = np.diag([0.25, (np.cos(theta[0]) ** 2) / 4])
assert np.allclose(res, exp)

theta = new_params
new_params = opt.step(circuit, theta)
```

File: tests/optimize/test_qng.py
Comment (josh146) [inline tests/optimize/test_qng.py:340]: 💯

**Review by josh146 — APPROVED** [review]
Agree. Thanks Antal!

**Pull request merged**
